File: simbot-spring-boot-starter-example/spring-boot-web-java/src/main/java/simbot/example/listener/ReplyListener.java
package simbot.example.listener;

import lombok.RequiredArgsConstructor;
import love.forte.simboot.annotation.Listener;
import love.forte.simbot.definition.Friend;
import love.forte.simbot.event.EventResult;
import love.forte.simbot.event.FriendMessageEvent;
import org.springframework.stereotype.Service;
import simbot.example.service.ReplyService;

/**
 * simbot中的监听类
 *
 * @author ForteScarlet
 */
@Service
@RequiredArgsConstructor
public class ReplyListener {

    private final ReplyService replyService;


    /**
     * 参数中使用 {@link FriendMessageEvent}, 代表监听好友消息事件。
     * 注意观察包路径，是 {@code love.forte.simbot.event.FriendMessageEvent }.
     *
     * @param friendMessageEvent 好友消息事件
     * @return 事件结果，用于控制事件流程。
     * 如果你不打算控制整个监听流程，返回值也可以直接使用 void .
     */
    @Listener
    public EventResult reply(FriendMessageEvent friendMessageEvent) {
        final String plainText = friendMessageEvent.getMessageContent().getPlainText().trim();

        final String reply = replyService.reply(plainText);
        if (reply == null) {
            // 如果没有此消息没有对应的回应消息，跳过
            return EventResult.defaults();
        }

        // 如果有，发送消息，并阻止后续事件的执行。
        final Friend currentFriend = friendMessageEvent.getFriend();
        currentFriend.sendBlocking(reply);

        // 返回 EventResult.truncate 代表阻止后续其他监听函数的执行。
        return EventResult.truncate();

    }

}

File: simbot-spring-boot-starter-example/spring-boot-webflux-kotlin/src/main/kotlin/simbot/example/listener/ReplyListener.kt
package simbot.example.listener

import love.forte.simboot.annotation.Listener
import love.forte.simbot.event.EventResult
import love.forte.simbot.event.FriendMessageEvent
import love.forte.simbot.event.inFriend
import org.springframework.stereotype.Service
import simbot.example.service.ReplyService

/**
 * simbot中的监听类
 *
 * @author ForteScarlet
 */
@Service
class ReplyListener(private val replyService: ReplyService) {

    /**
     * 参数中使用 [FriendMessageEvent], 代表监听好友消息事件。
     * 注意观察包路径，是 `love.forte.simbot.event.FriendMessageEvent` .
     *
     * @param friendMessageEvent 好友消息事件
     * @return 事件结果，用于控制事件流程。
     */
    @Listener
    suspend fun reply(friendMessageEvent: FriendMessageEvent): EventResult {
        val plainText = friendMessageEvent.messageContent.plainText.trim()
        // 如果没有可回复内容，不做任何处理。
        val reply = replyService.reply(plainText) ?: return EventResult.defaults()

        // 如果有，发送回复消息，并阻止后续事件的执行。
        friendMessageEvent.inFriend {
            send(reply)
        }

        return EventResult.truncate()
    }
}